test/models/ht_approval_request_test.rb

```diff
@@ -45,6 +45,22 @@ class HTApprovalRequestTest < ActiveSupport::TestCase
 
     assert_equal(HTApprovalRequest.digest(token), req.token_hash)
   end
+
+  test 'has correct Checkpoint attributes' do
+    req = create(:ht_approval_request, :expired)
+    assert_equal :ht_approval_request, req.resource_type
+    assert_equal req.id, req.resource_id
+  end
+
+  test 'for_user finds most recent or most incomplete' do
```

Comment:
How/where are we relying on this behavior? It's a little bit surprising to me for this behavior to be included with `for_user` -- something like this might better expose the intention:

```ruby
def self.most_recent(user)
  for_user(user).order(...)
end
```

Reply:
This is the first time we've explicitly allowed multiple requests for a single user, so this is grabbing the most recent and/or most NULL-heavy) ones.